src/main/java/com/redhat/pantheon/sling/ServiceResourceResolverProvider.java

```diff
@@ -0,0 +1,38 @@
+package com.redhat.pantheon.sling;
+
+import org.apache.sling.api.resource.LoginException;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import java.util.function.Consumer;
+
+@Component(
+        service = ServiceResourceResolverProvider.class
+)
+public class ServiceResourceResolverProvider {
+
+    private ResourceResolverFactory resourceResolverFactory;
+
+    @Activate
+    public ServiceResourceResolverProvider(
+            @Reference ResourceResolverFactory resourceResolverFactory) {
+        this.resourceResolverFactory = resourceResolverFactory;
+    }
+
+    public ResourceResolver getServiceResourceResolver() {
+        try {
+            return resourceResolverFactory.getServiceResourceResolver(null);
+        } catch (LoginException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void withServiceResourceResolver(Consumer<ResourceResolver> function) {
+        final ResourceResolver serviceResourceResolver = this.getServiceResourceResolver();
+        function.accept(serviceResourceResolver);
```

Comment:
Yes, this was my attempt at a functional API. Something that you could call like this:

```java
withServiceResourceResolver(resolver -> {
    // Use the service resolver to do stuff here
    resolver.createSomeNodes();
});
```
And the method would make sure to close the resources afterwards.
I desisted because of exception handling, but with more time I'd be able to make it work. Let me know if you'd rather not have the method.

Reply:
By all means, let's leave it in. I'm interested to see the technique. :)